src/Bicep.LangServer/Handlers/BicepCodeActionHandler.cs

```diff
@@ -39,15 +42,43 @@ public override Task<CommandOrCodeActionContainer> Handle(CodeActionParams reque
                 ? PositionHelper.GetOffset(compilationContext.LineStarts, request.Range.End)
                 : requestStartOffset;
 
-            var quickFixes = compilationContext.Compilation.GetEntrypointSemanticModel().GetAllDiagnostics()
+            var compilation = compilationContext.Compilation;
+            var diagnostics = compilation.GetEntrypointSemanticModel().GetAllDiagnostics();
+
+            var quickFixes = diagnostics
                 .Where(fixable =>
                     fixable.Span.ContainsInclusive(requestStartOffset) ||
                     fixable.Span.ContainsInclusive(requestEndOffset) ||
                     (requestStartOffset <= fixable.Span.Position && fixable.GetEndPosition() <= requestEndOffset))
                 .OfType<IFixable>()
                 .SelectMany(fixable => fixable.Fixes.Select(fix => CreateQuickFix(request.TextDocument.Uri, compilationContext, fix)));
 
-            return Task.FromResult(new CommandOrCodeActionContainer(quickFixes));
+            List<CommandOrCodeAction> commandOrCodeActions = new();
+
+            commandOrCodeActions.AddRange(quickFixes);
+
+            var analyzerDiagnostics = diagnostics
+                .Where(analyzerDiagnostic =>
+                    analyzerDiagnostic.Span.ContainsInclusive(requestStartOffset) ||
+                    analyzerDiagnostic.Span.ContainsInclusive(requestEndOffset) ||
+                    (requestStartOffset <= analyzerDiagnostic.Span.Position && analyzerDiagnostic.GetEndPosition() <= requestEndOffset))
+                .OfType<AnalyzerDiagnostic>()
+                .Select(analyzerDiagnostic => DisableLinterRule(documentUri, analyzerDiagnostic.Code, compilation.ConfigHelper.CustomSettingsFileName));
+
+            commandOrCodeActions.AddRange(analyzerDiagnostics);
+
+            return Task.FromResult(new CommandOrCodeActionContainer(commandOrCodeActions));
+        }
+
+        private static CommandOrCodeAction DisableLinterRule(DocumentUri documentUri, string ruleName, string? bicepConfigFilePath)
+        {
+            var command  = Command.Create(LanguageConstants.DisableLinterRuleCommandName, documentUri, ruleName, bicepConfigFilePath ?? string.Empty);
```

Comment:
>?? string.Empty [](http://example.com/codeflow?start=133&length=15)

Is it possible to pass `null` rather than an empty string to a command?

Reply:
I think it cannot be null https://github.com/OmniSharp/csharp-language-server-protocol/blob/112f24001e9d7c7929c60fd92dcaa3ca2025ba1e/src/Protocol/Models/Command.cs#L36

Reply:
Sounds good.